### PR TITLE
Lifetime cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
The last commit is very opinionated, let's see how we go.

r? @oli-obk

### CODE REVIEW

File: compiler/rustc_passes/src/loops.rs [hunk 1]
```diff
@@ -64,8 +64,8 @@ impl fmt::Display for BreakContextKind {
 }
 
 #[derive(Clone)]
-struct CheckLoopVisitor<'a, 'tcx> {
-    sess: &'a Session,
+struct CheckLoopVisitor<'tcx> {
+    sess: &'tcx Session,
```
Comment (oli-obk) [inline compiler/rustc_passes/src/loops.rs:68]: We can get the session from the tcx, the field can be removed

**Review by oli-obk — APPROVED** [review]

**Review by lcnr — COMMENTED** [review]
r=me after oli's review.

I feel like this change is right at the edge of whether it's worth it given the potential merge conflicts, reviewer time and what not. So while I think it's an improvement and we should merge this PR, I would not like to review many more PRs like this.

File: compiler/rustc_passes/src/loops.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_passes/src/loops.rs]: I've added a commit doing that.

**Comment by nnethercote** [comment]
> more PRs like this

The good news is that this PR does most of the these sorts of possible changes :)

**Comment by nnethercote** [comment]
FWIW, how I got to this PR:
- I noticed some convoluted memory management in `rustc_mir_transform`. I cleaned this up in #129720. One part of this change reduced the number of lifetimes for `FilterInformation` from four to two.
- This got me looking for other unnecessary lifetimes. In #130022 I removed unnecessary lifetimes from multiple widely-used structs in dataflow/borrowck code.
- I also noticed that `rustc_resolver` was using `'a` for all references into the `ResolverArenas` type. This is a situation that is perfect for a more descriptive name, so in #130208 I introduced the `'ra` lifetime name.
- This PR is the tail end of this work, where I found a few more unnecessary lifetimes and fixed up some lifetime ordering and naming inconsistencies. I didn't address every single possible inconsistency, just the more obvious ones.

**Pull request merged**